ha/core/event_analyzer/filter/filter.py

```diff
@@ -86,3 +102,54 @@ def filter_event(self, msg: str) -> bool:
         except Exception as e:
             raise EventAnalyzerError(f"Failed to filter event. Message: {msg}, Error: {e}")
 
+
+class IEMFilter(Filter):
+    """ Filter IEM consumed by watcher """
+
+    def __init__(self):
+        """
+        Init method
+        """
+        super().__init__()
+
+        # Get filter type and resource types list from the IEM rule file
+        self.filter_type = Conf.get(const.ALERT_FILTER_INDEX, "iem.filter_type")
+        self.components_list = Conf.get(const.ALERT_FILTER_INDEX, "iem.components")
+        self.modules_dict = Conf.get(const.ALERT_FILTER_INDEX, "iem.modules")
+
+    def filter_event(self, msg: str) -> bool:
+        """
+        Filter event.
+        Args:
+            msg (str): Msg
+        """
+        try:
+            iem_required = False
+            message = json.loads(msg)
+
+            _actuator_response_type = message.get('actuator_response_type')
+            if _actuator_response_type is not None:
+                return iem_required
+
+            _msg_type = Filter.get_msg_type(message)
+            if _msg_type != MESSAGETYPE.IEM.value:
+                return iem_required
+
+            _sensor_response_type = message.get('sensor_response_type')
+            _component_type = _sensor_response_type['specific_info']['module']
+            _module_type = _sensor_response_type['specific_info']['module']
+
+            if self.filter_type == const.INCLUSION:
```

Comment:
if self.filter_type == const.INCLUSION:

if self.filter_type == const.EXCLUSION:

both must be if else